Production_scripts/Process_extracts/extract_alarm_telecare.R

```diff
@@ -0,0 +1,228 @@
+    social_care_id = if_else(changed_sc_id, latest_sc_id, social_care_id)
+  ) %>%
+  ungroup()
+
+# Deal with episodes which have a package across quarters.
+qtr_merge <- at_full_clean %>%
+  # Use lazy_dt() for faster running of code
+  dtplyr::lazy_dt() %>%
+  group_by(
+    sending_location,
+    social_care_id,
+    record_keydate1,
+    smrtype,
+    period
+  ) %>%
+  # Create a count for the package number across episodes
+  mutate(pkg_count = row_number()) %>%
+  # Sort prior to merging
+  arrange(.by_group = TRUE) %>%
+  # group for merging episodes
+  group_by(
+    sending_location,
+    social_care_id,
+    record_keydate1,
+    smrtype,
+    pkg_count
+  ) %>%
+  # merge episodes with packages across quarters
+  # drop variables not needed
+  summarise(
+    sending_location = last(sending_location),
+    social_care_id = last(social_care_id),
+    sc_latest_submission = last(period),
+    record_keydate1 = last(record_keydate1),
+    record_keydate2 = last(record_keydate2),
+    smrtype = last(smrtype),
+    pkg_count = last(pkg_count),
+    chi = last(chi),
+    gender = last(gender),
+    dob = last(dob),
+    postcode = last(postcode),
+    recid = last(recid),
+    person_id = last(person_id),
+    sc_send_lca = last(sc_send_lca)
+  ) %>%
+  # sort after merging
+  arrange(
+    sending_location,
+    social_care_id,
+    record_keydate1,
+    smrtype,
+    sc_latest_submission
+  ) %>%
+  # end of lazy_dt()
+  as_tibble() %>%
+  # Sort for running SPSS
+  arrange(
+    sending_location,
+    social_care_id
+  )
+
+
+# Transform B17 ----
+
+# Now select epsiodes for given FY
+
+# Is there any function for fetch this address or should we write one?
+client_table_path <- fs::path(
+  glue::glue(
+    "/conf/sourcedev/Source_Linkage_File_Updates/{year}/Client_for_Source-20{year}"
+  ),
+  ext = "zsav"
+)
+client_table <-
+  haven::read_spss(client_table_path) %>% as.data.frame()
+
+final_out <- qtr_merge %>%
+  filter((
+    .data$record_keydate1 >= start_fy(year) &
+      .data$record_keydate2 <= end_fy(year)
+  ) |
+    (
+      .data$record_keydate1 <= end_fy(year) &
+        (
+          .data$record_keydate2 >= start_fy(year) |
+            is_missing(as.character(.data$record_keydate2))
+        )
+    )) %>%
+  as.data.frame() %>%
+  left_join(client_table, by = c("sending_location", "social_care_id")) %>%
+  arrange(chi, record_keydate1, record_keydate2)
+
+final_out %>%
+  select(
+    recid,
+    smrtype,
+    chi,
+    dob,
+    gender,
+    postcode,
+    sc_send_lca,
+    record_keydate1,
+    record_keydate2,
+    person_id,
+    sc_latest_submission,
+    sc_living_alone,
+    sc_support_from_unpaid_carer,
+    sc_social_worker,
+    sc_type_of_housing,
+    sc_meals,
+    sc_day_care
+  ) %>%
+  haven::write_sav(fs::path(
+    glue::glue("/conf/sourcedev/Source_Linkage_File_Updates/{year}/Alarms-Telecare-for-source-20{year}"),
+    ext = "zsav"
+  ))
```

Comment:
You will need to add the correct path for the alarm and telecare extract to the `get_source_extract_path` function.
```suggestion
  ) %>%
  write_rds(get_source_extract_path(year, type = "alarm_telecare", check_mode = "write"))
```